src/components/Map.js

```diff
@@ -48,12 +48,24 @@ map.addControl(
     map.on("load", () => {
       setTheMap(map);
       map.on("click", "parcels-fill", function (e) {
-        let parcel = map.queryRenderedFeatures(e.point, {
+        let results = map.queryRenderedFeatures(e.point, {
           layers: ["parcels-fill"],
         });
-        console.log(parcel)
-        setParcel(parcel[0].properties.parcelno);
-        history.push(`/${parcel[0].properties.parcelno}/`);
+        console.log(results)
+
+        // Subsequent clicks on the same shape cycle between overlapping parcels.
+        setParcel((currentParcel) => {
+          const parcels = results.map((p) => p.properties.parcelno)
```

Comment:
Note that I use the function-based version of `setParcel` here because the value that we want to set for the next `parcel` depends on the current value of `parcel`. Using the function-based `setParcel` provides the guarantee that `currentParcel` is up to date, whereas we'd have no such guarantee if we just used the prop `parcel`. (In fact, I was finding that it _was_ slightly out of date, which resulted in the next `parcel` being the same as the current one, so the concern was more than theoretical.)